scripts/commit-msg

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+set -o errexit
+
+exit_() {
+    echo ""
+    echo "$1"
+    echo ""
+    echo "This hook can be skipped if needed with 'git commit --no-verify'"
+    echo "See '.git/hooks/commit-msg', installed from 'scripts/commit-msg'"
+    exit 1
+}
+
+sign_offs="$(grep '^Signed-off-by: ' "$1" || test $? = 1 )"
+
+if [ -z "$sign_offs" ]; then
```

Comment:
Does this pass shellcheck? We usually prefer the Bashism [`[[`](http://mywiki.wooledge.org/BashFAQ/031).

Reply:
Yes, I run shellcheck and it passes just fine even with the `-s bash` option. Is there a way to enforce stylistic preferences with it?

What does the "bashism" entail in this context? Using `[[` & `]]` and not quoting variables? Anything else?

Reply:
I guess not. But yeah `[` is an actual command (literally `which [` yields `/usr/bin/[`) but `[[` is just Bash syntax, so it's fewer process calls. Not that I really care about performance here.